Add default timeout at 10 seconds

krallin opened the pull request:
This will cause requests to time out if it takes longer than 10 seconds to establish a connection.

cc @fancyremarker

krallin commented:
Fails on JRuby. Might be using a different adapter. Either way, the functionality is working (the exception is a Faraday one). I'll update the PR.

fancyremarker commented:
It's also OK to remove JRuby from the build matrix. Earlier we wanted to support both JRuby and MRI everywhere to get some theoretical performance improvements from running our API servers on JRuby, but in practical tests it was not more performant.

krallin commented:
No worries; I fixed it (it was an easy fix). Turns out this is because the JRuby version they have on Travis is Ruby 1.9.3.  I also moved our builds to Travis container infrastructure (since the non-container infra was down).

--

As a side note: I've also filed an issue with Faraday to understand the rationale for connection timeouts not being reported as timeouts on recent versions of Ruby that know about `Net::OpenTimeout` (https://github.com/lostisland/faraday/issues/561), which prevents them from being retried automatically.

Depending on what comes up, I think it might make sense for us to consider changing adapters so that our requests automatically get retried (otherwise, we'd need to make a lot of changes to the internals of hyper_resource so that we can pass our own retry config to Faraday).

fancyremarker commented:
Merged and released.
